File: HW1/ShapeCollection.java
import java.util.*;

abstract class Shape implements Comparable<Shape>{

    protected String color;
    protected int area;

    public Shape(String color, int area){
        this.color=color;
        this.area=area;
    }

    public abstract void display();

    @Override
    public int compareTo(Shape s){
        if(this.area==s.area)
            return 0;
        else if(this.area>s.area)
            return 1;
        else
            return -1;
    }

    public void setColor(String color){
        this.color = color;
    }

    public String getColor(){
        return(this.color);
    }

    public void setArea(int area){
        this.area = area;
    }

    public int getArea(){
        return(this.area);
    }
}

class Triangle extends Shape {
    public Triangle(String color, int area) {
        super(color, area);
    }

    @Override
    public void display(){
        System.out.println("I am a " + this.color + " triangle with area " + this.area);
    }
}

class Circle extends Shape {
    public Circle(String color, int area) {
        super(color, area);
    }

    @Override
    public void display(){
        System.out.println("I am a " + this.color + " circle with area " + this.area);
    }
}

class Square extends Shape {
    public Square(String color, int area) {
        super(color, area);
    }

    @Override
    public void display(){
        System.out.println("I am a " + this.color + " square with area " +   this.area);
    }
}

public class ShapeCollection{

     public static void main(String []args){

        ArrayList<Shape> shapes = new ArrayList<Shape>();
        
        System.out.println("Creating a blue triangle with area 4");
        shapes.add(new Triangle("blue", 4));
        System.out.println("Creating a green square with area 3");
        shapes.add(new Square("green", 3));
        System.out.println("Creating a red circle with area 5");
        shapes.add(new Circle("red", 5));

        System.out.println("Sorting shapes...");
        Collections.sort(shapes);

        shapes.forEach(shape -> shape.display());
}

File: HW2/shapecollection.py
from abc import ABCMeta, abstractmethod

class Shape:
    __metaclass__ = ABCMeta             # allows Shape to behave like an abstract class

    def __init__(self, color, area):
        self.color = color
        self.area  = area

    def __lt__(self, other_shape):      # function used to compare Shape classes by area attribute
         return self.area < other_shape.area

    @abstractmethod
    def display(self):
        pass

class Triangle(Shape):

    def __init__(self, color, area):
        super(Triangle, self).__init__(color, area)  # calling constructor of parent class

    def display(self):
        print("I am a {} triangle with area {}".format(self.color, self.area))

class Square(Shape):

    def __init__(self, color, area):
        super(Square, self).__init__(color, area)

    def display(self):
        print("I am a {} square with area {}".format(self.color, self.area))

class Circle(Shape):

    def __init__(self, color, area):
        super(Circle, self).__init__(color, area)

    def display(self):
        print("I am a {} circle with area {}".format(self.color, self.area))



def main():

    shapes = []

    t = Triangle("blue", 4)
    print("Creating a blue triangle with area 4...")
    s = Square("green", 3)
    print("Creating a green square with area 3...")
    c = Circle("red", 5)
    print("Creating a red circle with area 5...")

    shapes.extend((t, s, c))
    shapes.sort()
    print("Sorting shapes...")

    for shape in shapes:
        shape.display()


if __name__ == "__main__":
    main()
